modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusTxTracingConfigurationTest.java

```diff
@@ -0,0 +1,234 @@
+import static org.apache.ignite.transactions.TransactionConcurrency.PESSIMISTIC;
+import static org.apache.ignite.transactions.TransactionIsolation.SERIALIZABLE;
+
+/**
+ * Tests for transaction tracing configuration.
+ */
+public class OpenCensusTxTracingConfigurationTest extends AbstractOpenCensusTracingTest {
+
+    @Test
+    public void testTxConfigurationSamplingRateNeverPreventsTxTracing() throws Exception {
+        IgniteEx client = startGrid("client");
+
+        client.tracingConfiguration().addConfiguration(
+            new TracingConfigurationCoordinates.Builder(Scope.TX).build(),
+            new TracingConfigurationParameters.Builder().withSamplingRate(SAMPLING_RATE_NEVER).build());
+
+        client.transactions().txStart(PESSIMISTIC, SERIALIZABLE).commit();
+
+        handler().flush();
+
+        java.util.List<SpanData> gotSpans = handler().allSpans()
+            .filter(span -> SpanType.TX.traceName().equals(span.getName()))
+            .collect(Collectors.toList());
+
+        assertTrue(gotSpans.isEmpty());
+    }
+
+    @Test
+    public void testTxConfigurationSamplingRateAlwaysEnablesTxTracing() throws Exception {
+        IgniteEx client = startGrid("client");
+
+        client.tracingConfiguration().addConfiguration(
+            new TracingConfigurationCoordinates.Builder(Scope.TX).build(),
+            new TracingConfigurationParameters.Builder().withSamplingRate(SAMPLING_RATE_ALWAYS).build());
+
+        client.transactions().txStart(PESSIMISTIC, SERIALIZABLE).commit();
+
+        handler().flush();
+
+        java.util.List<SpanData> gotSpans = handler().allSpans()
+            .filter(span -> SpanType.TX.traceName().equals(span.getName()))
+            .collect(Collectors.toList());
+
+        assertEquals(1, gotSpans.size());
+    }
+
+    @Test
+    public void testTxConfigurationSamplingRateHalfSamplesSomethingAboutHalfTransactions() throws Exception {
+        IgniteEx client = startGrid("client");
+
+        client.tracingConfiguration().addConfiguration(
+            new TracingConfigurationCoordinates.Builder(Scope.TX).build(),
+            new TracingConfigurationParameters.Builder().withSamplingRate(0.5).build());
+
+        final int txAmount = 10;
+
+        for (int i = 0; i < txAmount; i++)
+            client.transactions().txStart(PESSIMISTIC, SERIALIZABLE).commit();
+
+        handler().flush();
+
+        java.util.List<SpanData> gotSpans = handler().allSpans()
+            .filter(span -> SpanType.TX.traceName().equals(span.getName()))
+            .collect(Collectors.toList());
+
+        // Cause of probability nature of sampling it's not possible to check that 0.5 sampling rate will end with
+        // 5 sampling transactions out of {@code txAmount},
+        // so we just check that some and not all transactions were traced.
+        assertTrue(!gotSpans.isEmpty() && gotSpans.size() < txAmount);
```

Comment:
This test will have ~0.2% flaky rate.

Reply:
Agreed, let it be 100 transactions instead of 10.